### PR TITLE
executor: track memory usage of map in agg partial result.

### PR DESCRIPTION
**Pull request by wshwsh12**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:
https://github.com/pingcap/tidb/pull/22965 introduce set_with_memory_usage.
In the pr, I use it to trace the memory usage of set in aggregate.

How it Works:

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

- No release note

### CODE REVIEW

**Review by ichn-hu — COMMENTED** [review]
Rest LGTM

File: executor/aggfuncs/func_avg.go [hunk 1]
```diff
@@ -193,25 +193,26 @@ func (e *avgPartial4Decimal) MergePartialResult(sctx sessionctx.Context, src, ds
...
-	return PartialResult(p), DefPartialResult4AvgDistinctDecimalSize
+	return PartialResult(p), DefPartialResult4AvgDistinctDecimalSize + setSize
 }
 
 func (e *avgOriginal4DistinctDecimal) ResetPartialResult(pr PartialResult) {
 	p := (*partialResult4AvgDistinctDecimal)(pr)
 	p.sum = *types.NewDecFromInt(0)
 	p.count = int64(0)
-	p.valSet = set.NewStringSet()
+	p.valSet, _ = set.NewStringSetWithMemoryUsage()
```
[the remark this answers is not in the document]
Comment (wshwsh12) [reply executor/aggfuncs/func_avg.go:215]: ResetPartialResult is only called by StreamAgg/WindowFunction, and now we only consider StreamAgg.
I try to record the memory usage of initializating partial results. And when calling ResetPartialResult, we can reset the memory usage to the init values. Is it ok? @ichn-hu

[the remark this answers is not in the document]
Comment (ichn-hu) [reply executor/aggfuncs/func_avg.go:215]: Yeah, I think that works.

**Review by ichn-hu — APPROVED** [review]

File: executor/aggregate.go [hunk 2]
```diff
@@ -1053,7 +1061,8 @@ type StreamAggExec struct {
 	groupRows          []chunk.Row
 	childResult        *chunk.Chunk
 
-	memTracker *memory.Tracker // track memory usage.
+	memTracker                   *memory.Tracker // track memory usage.
+	initPartialResultMemoryUsage int64
```
Comment (XuHuaiyu) [inline executor/aggregate.go:1065]: What does this mean? Please add a comment for this

**Review by XuHuaiyu — APPROVED** [review]
/LGTM

**Pull request merged**